org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalReplacementProvider.java

```diff
@@ -135,6 +136,15 @@ public void updateReplacement(CompletionProposal proposal, CompletionItem item,
 		}
 
 		if (range == null) {
+			PreferenceManager preferenceManager = JavaLanguageServerPlugin.getPreferencesManager();
+			boolean completionOverwrite = preferenceManager == null || preferenceManager.getPreferences().isCompletionOverwrite();
+			if (completionOverwrite && (proposal.getKind() == CompletionProposal.LOCAL_VARIABLE_REF || proposal.getKind() == CompletionProposal.FIELD_REF)) {
```

Comment:
- it seems to me it should be `if (!completionOverwrite`
- I believe `proposal.getKind() == CompletionProposal.METHOD_REF` is missing in the comparison